### PR TITLE
Set experimental features to stable for 7.4 release

### PR DESCRIPTION
**Pull request by JamesWTruher**
# PR Summary

## PR Context

see https://github.com/orgs/PowerShell/discussions/20214

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 1]
```diff
@@ -4652,16 +4652,13 @@ static InitialSessionState()
...
-                        SpecialVariables.PSNativeCommandUseErrorActionPreference,
-                        value: true,    // when this feature is changed to stable, this should default to `false`
-                        RunspaceInit.PSNativeCommandUseErrorActionPreferenceDescription,
-                        ScopedItemOptions.None,
-                        new ArgumentTypeConverterAttribute(typeof(bool))));
-            }
+            builtinVariables.Add(
+                new SessionStateVariableEntry(
+                    SpecialVariables.PSNativeCommandUseErrorActionPreference,
+                    value: true,    // when this feature is changed to stable, this should default to `false`
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/InitialSessionState.cs:4658]: You missed this important comment :)

```suggestion
                    value: false,
```

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

File: src/System.Management.Automation/engine/BytePipe.cs [hunk 2]
```diff
@@ -104,9 +104,7 @@ internal static FileBytePipe Create(string fileName, bool append)
             throw new RuntimeException(null, e, errorRecord);
         }
 
-        ApplicationInsightsTelemetry.SendExperimentalUseData(
-            ExperimentalFeature.PSNativeCommandPreserveBytePipe,
-            "f");
+        ApplicationInsightsTelemetry.SendExperimentalUseData("PSNativeCommandPreserveBytePipe", "f");
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/BytePipe.cs:107]: `PSNativeCommandPreserveBytePipe` is no longer an experimental feature, so this telemetry call should be removed.

File: src/System.Management.Automation/engine/NativeCommandProcessor.cs [hunk 3]
```diff
@@ -547,7 +546,7 @@ private void InitNativeProcess()
 
             // Send Telemetry indicating what argument passing mode we are in.
             ApplicationInsightsTelemetry.SendExperimentalUseData(
-                ExperimentalFeature.PSWindowsNativeCommandArgPassing,
+                "PSWindowsNativeCommandArgPassing",
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/NativeCommandProcessor.cs:549]: hmm, this appears again, so it seems intentional. Why do we want to continue to send this telemetry?

File: src/System.Management.Automation/engine/NativeCommandProcessor.cs [hunk 4]
```diff
@@ -972,12 +955,12 @@ internal override void Complete()
                     // The variable is unset
                     if (useDefaultSetting)
                     {
-                        ApplicationInsightsTelemetry.SendExperimentalUseData(ExperimentalFeature.PSNativeCommandErrorActionPreferenceFeatureName, "unset");
+                        ApplicationInsightsTelemetry.SendExperimentalUseData("PSNativeCommandErrorActionPreference", "unset");
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/NativeCommandProcessor.cs:958]: same here

File: src/System.Management.Automation/engine/NativeCommandProcessor.cs [hunk 5]
```diff
@@ -972,12 +955,12 @@ internal override void Complete()
...
                         return;
                     }
 
                     // Send the value that was set.
-                    ApplicationInsightsTelemetry.SendExperimentalUseData(ExperimentalFeature.PSNativeCommandErrorActionPreferenceFeatureName, nativeErrorActionPreferenceSetting.ToString());
+                    ApplicationInsightsTelemetry.SendExperimentalUseData("PSNativeCommandErrorActionPreference", nativeErrorActionPreferenceSetting.ToString());
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/NativeCommandProcessor.cs:963]: Here too.

File: src/System.Management.Automation/engine/pipeline.cs [hunk 6]
```diff
@@ -261,31 +261,26 @@ private void LogToEventLog()
...
-                        {
-                            ApplicationInsightsTelemetry.SendExperimentalUseData(
-                                ExperimentalFeature.PSNativeCommandPreserveBytePipe,
-                                "p");
-                            _haveReportedNativePipeUsage = true;
-                        }
-
-                        _lastNativeCommand.DownStreamNativeCommand = nativeCommand;
-                        nativeCommand.UpstreamIsNativeCommand = true;
+                        ApplicationInsightsTelemetry.SendExperimentalUseData("PSNativeCommandPreserveBytePipe", "p");
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/pipeline.cs:271]: Same here

File: src/System.Management.Automation/engine/BytePipe.cs (hunk 2 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/BytePipe.cs:107]: We want to retain telemetry to monitor usage

File: src/System.Management.Automation/engine/NativeCommandProcessor.cs (hunk 3 above)
Comment (JamesWTruher) [reply src/System.Management.Automation/engine/NativeCommandProcessor.cs:549]: these features are still interesting to us to help us understand how the system is being used. We don't have another API to send usage telemetry, so we're continuing to use (and need) this.

File: src/System.Management.Automation/engine/NativeCommandProcessor.cs (hunk 4 above)
Comment (JamesWTruher) [reply src/System.Management.Automation/engine/NativeCommandProcessor.cs:958]: see above

File: src/System.Management.Automation/engine/NativeCommandProcessor.cs (hunk 5 above)
Comment (JamesWTruher) [reply src/System.Management.Automation/engine/NativeCommandProcessor.cs:963]: see above

**Review by SeeminglyScience — APPROVED** [review]
LGTM!

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**